f2/src/InformationPage.js

```diff
@@ -37,15 +41,19 @@ export const InformationPage = () => {
 
               <InformationForm
                 onSubmit={(data) => {
+                  navigate(doneForms, pages, (updatedPages) => {
+                    dispatch({
```

Comment:
I always feel that I am missing something in React. Do we really need to call dispatch? I understand that we have nav stage stored in global state. But the core value of dispatch is to re-render whoever access that value in other pages or other components in the same page. 
But in this case, we use following code immediately. It feels to me that we don't really need to dispatch since pages.nextPage.url is up to date.
history.push(pages.nextPage.url)

Reply:
Tell me if I miss some fundamentals. From time to time I feel so junior.

Reply:
You are correct. This was my misunderstanding - I thought we had to update state to reflect changes.